src/plots/cartesian/axes.js

```diff
@@ -2411,11 +2413,12 @@ function swapAxisGroup(gd, xIds, yIds) {
     for(i = 0; i < xIds.length; i++) xFullAxes.push(axes.getFromId(gd, xIds[i]));
     for(i = 0; i < yIds.length; i++) yFullAxes.push(axes.getFromId(gd, yIds[i]));
 
-    var allAxKeys = Object.keys(xFullAxes[0]),
-        noSwapAttrs = [
-            'anchor', 'domain', 'overlaying', 'position', 'side', 'tickangle'
-        ],
-        numericTypes = ['linear', 'log'];
+    var allAxKeys = Object.keys(axAttrs);
+
+    var noSwapAttrs = [
+        'anchor', 'domain', 'overlaying', 'position', 'side', 'tickangle', 'editType'
```

Comment:
> How can axAttrs with editType get here?

Because I'm using the schema instead of `fullLayout` to find the relevant attributes now. The reason for this change is that not all the attributes that should be swapped are always coerced - the one that prompted this change was `rangemode`, since I changed its behavior here, but there are others I'm sure.

It wouldn't be a big deal to swap a non-attribute like `editType`, but in principle we shouldn't, as we won't swap other non-attributes.

Of course it's not clear anyone is even using `Axes.swap` anymore but it's here 😐

Reply:
Makes sense :+1: